docs/winrt/WinRT SxS Components.md

```diff
@@ -1 +1,153 @@
-## Placeholder
\ No newline at end of file
+# Side-by-Side Versioned Components
+
+## Status
+Proposed: This document describes a proposed design that has not been implemented.
+
+## Background
+The Windows Runtime type system was largely designed with the needs of opertating system components in mind. Although applications can define their own API types using the Windows Runtime type system, the feature set for doing so is somewhat basic. Versioning strategies are only vaguely defined, and no provisions are made for side-by-side support of a component. As the community has developed more sophisticated libraries over time, these deficiencies are becoming more challenging, and further support is needed.
+
+Today, the Windows Runtime only supports binary-compatible versioning. That is, if a an application or component is developed with a dependency on version N of another component, version N+1 *can* be developed in such a way that code designed to use version N can leverage version N+1 without changes or recompilation of the client component. Alternately, a component author can ignore binary compatbility (which comes with trade-offs) and require that applications use a single, specific version of the component. 
+
+Up to this point, this has been largely sufficient. The single largest "component" is the Windows operating system, which has processes in place to ensure binary compatbility. Other components have typically been distributed as part of an MSIX packaged application. he expectation is that the application developer can manage the dependency graph. This has largely been true up to this point. However, authors of new components are recognizing that they will be consumed in more complex dependency graphs, where the application may have dependencies on multiple versions of the component, and the component author does not want to be locked into a binary-stable model. This implies the need for some form of side by side support.
```

Comment:
The versioning for Windows may be worth drilling into more. It has been a simple: Keep it binary compatible. If it cannot be binary compatible, give it a different name. I do not understand why this simple model does not work as a solution that this proposal is trying to solve.

Reply:
This will remain true for Windows. There are a growing number of Windows Runtime components that are shipped outside of windows as well. Owners of some of those components don't want to reset user codebases to make breaking changes across versions, and are willing to do the work to ensure side-by-side compatibility. This makes it easier to write client code that can easily be reused across versions with little or no changes. If you read carefully between the lines, this effectively is giving everything a new name under the hood, but providing the ability to paper over it for the sake improving code portability and mitigating the cost of version migration.

Binary compatibility offers many, many benefits, but for some scenarios there are good reasons why this may be unavoidable.

Reply:
>  If you read carefully between the lines, this effectively is giving everything a new name under the hood

Right, this is giving everything a new name using magic. My point is that you can just tell people to give things new name when they choose to break, and do not build the magic.